### ISSUE TITLE
company-coq is 2x slower than vanilla PG

### ISSUE THREAD

**Issue by JasonGross** [cpitclaudel/company-coq#137]
If I go to the end of the line at https://github.com/mit-plv/fiat/blob/13fecc64d7ec3ae4f969f807c73b26d123020abc/src/Parsers/Refinement/IndexedAndAtMostOneNonTerminalReflective.v#L668 and execute `proof-goto-point`, vanilla PG takes 5-6 seconds, while PG + company-coq takes 10-12 seconds.  (This is consistent across ~5-10 executions of each.  GNU Emacs 24.5.1 + https://github.com/ProofGeneral/PG/commit/771cab48b2f9ea2ae3fa8f944d0e36a805bf9f3b)

**Comment by CarlOlson** [comment]
This is likely because company-coq uses overlays. Overlays are generally considered slow, there is probably no fix to this (other than refactoring out using overlays).

**Comment by cpitclaudel** [comment]
No, I don't think so: company-coq doesn't add overlays beyond what PG does, and overlays have typically only been a minor issue in PG (it may change in PG/XML, which uses them more heavily).  There is ongoing work in Emacs to make overlays much faster; see https://lists.gnu.org/archive/html/emacs-devel/2016-09/msg00582.html :)
